### PR TITLE
custom-set: Rewrite tests to use hspec with fail-fast.

### PR DESCRIPTION
**Pull request by rbasso**
- Rewrite tests to use hspec.
- Remove explicit testing of `Show`.

Related to #211.

### CODE REVIEW

File: exercises/custom-set/test/Tests.hs [hunk 1]
```diff
@@ -1,92 +1,204 @@
...
+        it "delete existing element" $
+          delete 2 (fromList [1, 2, 3]) `shouldBe` fromList [1, 3]
+
+      describe "size" $ do
+
+        it "size of an empty set is zero" $
+          size (fromList ([] :: [Integer])) `shouldBe` 0
+
+        it "size is the number of elements" $
+          size (fromList [1, 2, 3, 2]) `shouldBe` 3
```
Comment (petertseng) [inline exercises/custom-set/test/Tests.hs]: the same as the below test!

File: exercises/custom-set/test/Tests.hs [hunk 2]
```diff
@@ -1,92 +1,204 @@
...
+
+        it "size doesn't count repetitions in `fromList`" $
+          size (fromList [1, 2, 3, 2]) `shouldBe` 3
+
+        it "size doesn't count repetitions from `insert`" $
```
Comment (petertseng) [inline exercises/custom-set/test/Tests.hs]: I don't think this was a repetition though. I would have said repetition if it were ```size (insert 3 (fromList [1, 3])) `shouldBe` 2```

Comment (rbasso) [inline exercises/custom-set/test/Tests.hs]: Thank you very much for the review! The `size` tests where really wrong! :smile:

I think it is fixed now. :+1: :question:

**Comment by petertseng** [comment]
:+1:

**Pull request merged**